crypto/ec/ec_ameth.c

```diff
@@ -520,6 +520,19 @@ static int ec_pkey_ctrl(EVP_PKEY *pkey, int op, long arg1, void *arg2)
 
 }
 
+static int ec_pkey_check(const EVP_PKEY *pkey)
+{
+    EC_KEY *eckey = pkey->pkey.ec;
+
+    /* stay consistent to what EVP_PKEY_check demands */
+    if (eckey == NULL || eckey->group == NULL || eckey->pub_key == NULL
+        || eckey->priv_key == NULL) {
```

Comment:
Actually, you only need to check `eckey->priv_key`...  and you should generate an error...  something like this?

``` C
        ECerr(EC_F_EC_PKEY_CHECK, EC_R_NOT_A_PRIVATE_KEY);
```

Reply:
Exactly...